### DISCUSSION TITLE
question: Cancel/abort async validation

### DISCUSSION THREAD

**Discussion by TheCelavi** [typestack/class-validator#2368] [Questions & Answers]
I have tried to find any information on how to cancel validation and/or how to create custom cancelable async validator.

When dealing with canceling, RxJS (observables) are way to go, but since this library does not support observables (which is understandable), support for abort signal should be sufficient enough. Especially if async validator contacts server through network for validation (per example, unique email where check is done while user types e-mail address, with some debouncing).

So, question is:
- Is abort signal supported? (better yet, am I blind 😄 )
- If abort signal is not supported, are there any plans to provide support?

Thank you!

**Comment by braaar** [comment]
I'm moving this to the discussion tab, as we are trying to keep the issue count low. Please see [this issue](https://github.com/typestack/class-validator/issues/1775) for an explanation.
